bugsnag/bugsnag-android
# Update Event interface to match notifier spec

## Goal

Updates the `Event` interface to use the new interfaces such as `Error`, and to be named consistently as per the notifier spec.

Note: this PR relies on #668 

## Changeset

- Remove `BugsnagException` as this interface is now unused
- Remove `Exceptions` as this implementation detail is now unused in favour of `Error`
- Add a method to the `BugsnagPluginInterface` for constructing an ANR event, as the ANR plugin cannot access the internally visible `Event` constructor directly
- Update `NativeInterface.notify()` to call `updateSeverityInternal()`, which fixes a pre-existing bug where native notifies show the wrong severityReason (userCallback)
- Update `ObjectJsonStreamer` to serialize a `Streamable` object directly, fixing a `Stackframe` serialization bug with native handled errors
- Update test code to use new `Event` constructor and renamed fields
- Convert `NullMetadataTest` to JVM test
- Update `EventSerializationTest` to parameterized JVM test. Note that test cases related to `HandledState` serialization have been removed as they already exist in `HandledStateSerializationTest`.
- Update `Client` to use new `Event` constructor and renamed fields

### Event
- Converted `Event` interface to Kotlin
- Renamed `appData` to `app`
- Renamed `deviceData` to `device`
- Added `originalError`
- Made existing properties public where appropriate according to the notifier spec
